lib/service/service.go

```diff
@@ -3752,17 +3746,50 @@ func (process *TeleportProcess) initDiagnosticService() error {
 }
 
 // initDebugService starts debug service serving endpoints used for
-// troubleshooting the instance.
-func (process *TeleportProcess) initDebugService() error {
+// troubleshooting the instance. This service is always active, users can
+// disable its sensitive pprof and log-setting endpoints, but the liveness
+// and readiness ones are always active.
+func (process *TeleportProcess) initDebugService(exposeDebugRoutes bool) error {
 	logger := process.logger.With(teleport.ComponentKey, teleport.Component(teleport.ComponentDebug, process.id))
 
+	// Unix socket creation can fail on paths too long. Depending on the UNIX implementation,
+	// socket paths cannot exceed 104 or 108 chars.
 	listener, err := process.importOrCreateListener(ListenerDebug, filepath.Join(process.Config.DataDir, teleport.DebugServiceSocketName))
 	if err != nil {
-		return trace.Wrap(err)
+		// If the debug service is enabled in the config, this is a hard failure
+		if exposeDebugRoutes {
```

Comment:
I'll do a quick check to make sure it doesn't reintroduce the problem fixed by https://github.com/gravitational/teleport/pull/43267, but from what I'm reading here it seems like it shouldn't.

Reply:
Yeah, everything works fine, I see the warning, but the agent starts and joins the cluster.

Reply:
Thank you for checking!